pkg/applicationserver/grpc_deviceregistry.go

```diff
@@ -19,20 +19,49 @@ import (
 
 	pbtypes "github.com/gogo/protobuf/types"
 	"go.thethings.network/lorawan-stack/pkg/auth/rights"
+	"go.thethings.network/lorawan-stack/pkg/crypto/cryptoutil"
 	"go.thethings.network/lorawan-stack/pkg/errors"
 	"go.thethings.network/lorawan-stack/pkg/ttnpb"
 )
 
 type asEndDeviceRegistryServer struct {
-	registry DeviceRegistry
+	AS *ApplicationServer
 }
 
 // Get implements ttnpb.AsEndDeviceRegistryServer.
 func (r asEndDeviceRegistryServer) Get(ctx context.Context, req *ttnpb.GetEndDeviceRequest) (*ttnpb.EndDevice, error) {
 	if err := rights.RequireApplication(ctx, req.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_DEVICES_READ); err != nil {
 		return nil, err
 	}
-	return r.registry.Get(ctx, req.EndDeviceIdentifiers, req.FieldMask.Paths)
+	dev, err := r.AS.deviceRegistry.Get(ctx, req.EndDeviceIdentifiers, req.FieldMask.Paths)
+	if err != nil {
+		return nil, err
+	}
+	for _, s := range []struct {
```

Comment:
IMO this is overly complex for what it does
```go
if ttnpb.HasAnyField(req.FieldMask.Paths, "session.keys.app_s_key") && dev.GetSession().GetAppSKey() != nil {
	key, err := cryptoutil.UnwrapAES128Key(*dev.Session.AppSKey, r.AS.KeyVault)
	if err != nil {
		return nil, err
	}
	dev.Session.AppSKey = &ttnpb.KeyEnvelope{Key: &key}
}

if ttnpb.HasAnyField(req.FieldMask.Paths, "pending_session.keys.app_s_key") && dev.GetPendingSession().GetAppSKey() != nil {
	key, err := cryptoutil.UnwrapAES128Key(*dev.PendingSession.AppSKey, r.AS.KeyVault)
	if err != nil {
		return nil, err
	}
	dev.PendingSession.AppSKey = &ttnpb.KeyEnvelope{Key: &key}
}
```

Is simpler, shorter and more efficient. Please consider simplifying this.

Reply:
Yeah this comes from copy-pasting from NS and we wanted to keep these gRPC device registry implementations in sync to a large extent, but I agree it's too extreme in this case.

Reply:
This is obviously my responsibility but your suggestion was overly simple, https://github.com/TheThingsNetwork/lorawan-stack/issues/853